exec/cnex/httpserver.h

```diff
@@ -0,0 +1,53 @@
+#ifndef HTTPSERVER_H
+#define HTTPSERVER_H
+#include <assert.h>
+
+#include "nstring.h"
+
+#include "lib/socketx.h"
+
+struct tagTExecutor;
+
+typedef struct tagTHttpResponse {
+    unsigned int code;
+    TString *content;
+} HttpResponse;
+
+HttpResponse *response_createResonse();
+void response_freeResponse(HttpResponse *r);
+
+typedef struct tagTIHttpServerHandler {
+    void (*handle_GET)(struct tagTExecutor *exec, TString *path, struct tagTHttpResponse *response);
+    void (*handle_POST)(struct tagTExecutor *exec, TString *path, TString *data, struct tagTHttpResponse *response);
+} IHttpServerHandler;
+
+typedef struct tagTClient {
+    struct tagTExecutor *exec;
+    IHttpServerHandler *handler;
+    SOCKET socket;
+    TString *request;
+    TString *path;
+    int64_t post_length;
+    TString *post_data;
+} Client;
+
+Client *client_createClient(struct tagTExecutor *exec, struct tagTIHttpServerHandler *handler, SOCKET socket);
```

Comment:
All these functions (and also the Client structure declaration above) don't seem to be part of the public interface and so don't need to be present in this file. You can probably remove some other things from this header file too, like `HttpResponse`. And that might een remove the need to include `"nstring.h"` here.

Reply:
Cleaned up header file, and fixed up dependencies.  This touched a couple more files, but nothing major.